pkg/cloud/services/networking/router.go

```diff
@@ -239,19 +235,76 @@ func (s *Service) getRouterInterfaces(routerID string) ([]ports.Port, error) {
 	return portList, nil
 }
 
-func (s *Service) existsRouter(routerID string) (bool, error) {
-	if routerID == "" {
-		return false, nil
+type Router struct {
+	router routers.Router
+	subnet subnets.Subnet
+}
+
+func (s *Service) getRouter(clusterName string) (*Router, error) {
```

Comment:
please return the Router and Subnet in individual vars. In Go imho usually there are no struct used to just wrap return values:
```go
(routers.Router, subnet.Subnet, error)
```